qiskit_superstaq/compiler_output.py

```diff
@@ -31,8 +31,8 @@ def __init__(
         ],
         pulse_sequences: Union[qiskit.pulse.Schedule, List[qiskit.pulse.Schedule]] = None,
         seq: Optional["qtrl.sequencer.Sequence"] = None,
-        jaqal_programs: Optional[List[str]] = None,
-        pulse_lists: Optional[Union[List[List], List[List[List]]]] = None,
+        jaqal_programs: Optional[Union[str, List[str]]] = None,
+        pulse_lists: Optional[Union[List[List[str]], List[List[List[str]]]]] = None,
```

Comment:
should have one more layer of Lists, and the pulses themselves are qtrl UniquePulses (so should be `Any` to avoid requiring qtrl for type checks)
```suggestion
        pulse_lists: Optional[Union[List[List[List[Any]]], List[List[List[List[Any]]]]]] = None,
```
or if we want to be even more specific, qtrl pulse types are actually namedtuples i *think* this should work: 
```suggestion
        pulse_lists: Optional[Union[List[List[List[AQTPulseType]]], List[List[List[List[AQTPulseType]]]]]] = None,
```
where
```
AQTPulseType = Tuple[Tuple[Any, ...], float, int, int]
```